experimental/sev_guest/src/io.rs

```diff
@@ -165,43 +166,118 @@ where
 /// Factory for creating port reader and writers that perform direct raw IO operations.
 pub struct RawIoPortFactory;
 
-impl<'a, T> IoPortFactory<'a, T, PortReadOnly<T>, PortWriteOnly<T>> for RawIoPortFactory
+impl<'a, T> IoPortFactory<'a, T, Port<T>, Port<T>> for RawIoPortFactory
 where
     T: 'a,
-    PortReadOnly<T>: PortReader<T>,
-    PortWriteOnly<T>: PortWriter<T>,
+    Port<T>: PortReader<T>,
+    Port<T>: PortWriter<T>,
 {
-    fn new_reader(&self, port: u16) -> PortReadOnly<T> {
-        PortReadOnly::<T>::new(port)
+    fn new_reader(&self, port: u16) -> Port<T> {
+        Port::<T>::new(port)
     }
 
-    fn new_writer(&self, port: u16) -> PortWriteOnly<T> {
-        PortWriteOnly::<T>::new(port)
+    fn new_writer(&self, port: u16) -> Port<T> {
+        Port::<T>::new(port)
     }
 }
 
-impl PortReader<u8> for PortReadOnly<u8> {
+impl PortReader<u8> for Port<u8> {
     unsafe fn try_read(&mut self) -> Result<u8, &'static str> {
         Ok(self.read())
     }
 }
 
-impl PortReader<u32> for PortReadOnly<u32> {
+impl PortReader<u32> for Port<u32> {
     unsafe fn try_read(&mut self) -> Result<u32, &'static str> {
         Ok(self.read())
     }
 }
 
-impl PortWriter<u8> for PortWriteOnly<u8> {
+impl PortWriter<u8> for Port<u8> {
     unsafe fn try_write(&mut self, value: u8) -> Result<(), &'static str> {
         self.write(value);
         Ok(())
     }
 }
 
-impl PortWriter<u32> for PortWriteOnly<u32> {
+impl PortWriter<u32> for Port<u32> {
     unsafe fn try_write(&mut self, value: u32) -> Result<(), &'static str> {
         self.write(value);
         Ok(())
     }
 }
+
+/// An IO port reader and writer implementation that uses the GHCB protocol, static references and a
```

Comment:
"GHCB protocol" - is this the GHCB IOIO protocol? Do we have a reference to the GHBC IOIO protocol?

Reply:
Yes, this is the GHCB IOIO protocol. Updated the comment to clarify this. We have links to the GHCB IOIO protocol specification in the `ghcb` module where `Ghcb` and `GhcbProtocol` are defined. I don't think we need to add those links to every comment where the protocol is mentioned.